apps/test/netsim/NetSimSendPanel.js

```diff
@@ -0,0 +1,102 @@
+var MessageGranularity = NetSimConstants.MessageGranularity;
+
+/**
+ * Simulate an enter key press when the given element is in focus.
+ * @param {jQuery} jQueryElement
+ */
+function simulateEnterKeyPress(jQueryElement) {
+  var e = $.Event('keydown');
+  e.which = KeyCodes.ENTER;
+  e.keyCode = KeyCodes.ENTER;
+  jQueryElement.trigger(e);
+}
+
+describe("NetSimSendPanel", function () {
+  var testShard, localNode, remoteNode;
+  var panel, rootDiv, stubNetSim;
+
+  beforeEach (function () {
+    NetSimTestUtils.initializeGlobalsToDefaultValues();
+    NetSimGlobals.getLevelConfig().defaultEnabledEncodings = ['binary'];
+
+    testShard = NetSimTestUtils.fakeShard();
+    NetSimLocalClientNode.create(testShard, 'Local Lois', function (_, node) {
+      localNode = node;
+    });
+    NetSimLocalClientNode.create(testShard, 'Remote Ralph', function (_, node) {
+      remoteNode = node;
+    });
+    assert(localNode && remoteNode, "Created test nodes");
+
+    localNode.connectToClient(remoteNode, function () {});
+    remoteNode.connectToClient(localNode, function () {});
+
+    rootDiv = $('<div>');
+    stubNetSim = {
+      animateSetWireState: function () {},
+      myNode: localNode,
+      runLoop_: {
+        tick: {
+          register: function () {}
+        }
+      },
+      updateLayout: function () {}
+    };
+  });
+
+  describe ("in single-bit mode", function () {
+    beforeEach (function () {
+      NetSimGlobals.getLevelConfig().messageGranularity = MessageGranularity.BITS;
+      panel = new NetSimSendPanel(rootDiv, NetSimGlobals.getLevelConfig(), stubNetSim);
+    });
+
+    it ("sends a single bit on 'Set Wire' button click", function () {
+      panel.packets_[0].setPacketBinary('1000');
+      panel.getBody().find('#set-wire-button').click();
+      assert.equal('000', panel.packets_[0].getPacketBinary());
+    });
+
+    it ("sends a single bit on pressing enter", function () {
+      panel.packets_[0].setPacketBinary('1000');
+      simulateEnterKeyPress(rootDiv.find('textarea.message'));
+      assert.equal('000', panel.packets_[0].getPacketBinary());
+    });
+  });
+
+  describe ("in packet mode", function () {
+    beforeEach (function () {
+      NetSimGlobals.getLevelConfig().messageGranularity = MessageGranularity.PACKETS;
+      panel = new NetSimSendPanel(rootDiv, NetSimGlobals.getLevelConfig(), stubNetSim);
+    });
+
+    it ("sends all packets on 'Send' button click", function () {
+      panel.packets_[0].setPacketBinary('1000');
+      panel.getBody().find('#send-button').click();
+      panel.tick({time:0});
+      assert.equal('', panel.packets_[0].getPacketBinary());
+    });
+
+    it ("sends all packets on pressing enter", function () {
+      panel.packets_[0].setPacketBinary('1000');
+      simulateEnterKeyPress(rootDiv.find('textarea.message'));
+      panel.tick({time:0});
+      assert.equal('', panel.packets_[0].getPacketBinary());
+    });
+  });
+});
```

Comment:
I think our style guide advocates no space after describe/it https://github.com/code-dot-org/code-dot-org/blob/staging/STYLEGUIDE.md#js-parentheses-spacing